lib/perso.ml

```diff
@@ -2615,141 +2603,73 @@ and eval_person_field_var conf base env ((p, p_auth) as ep) loc = function
       | Vint cnt -> VVstring (string_of_int cnt)
       | _ -> VVstring "")
   | [ "cous_paths_min_date"; l1; l2 ] ->
-      let max_a_l =
-        match get_env "max_anc_level" env with
-        | Vint i -> i
-        | _ -> max_anc_level_default
-      in
-      let max_d_l =
-        match get_env "max_desc_level" env with
-        | Vint i -> i
-        | _ -> max_desc_level_default
-      in
       let _cousins_cnt, cousins_dates =
         match (!Cousins.cousins_t, !Cousins.cousins_dates_t) with
         | Some t, Some d_t -> (t, d_t)
-        | _, _ -> Cousins.init_cousins_cnt base max_a_l max_d_l p
+        | _, _ -> Cousins.init_cousins_cnt conf base p
       in
       let i = try int_of_string l1 with Failure _ -> raise Not_found in
       let j = try int_of_string l2 with Failure _ -> raise Not_found in
-      let min, _ = cousins_dates.(i).(j) in
+      let min, _ =
+        if
+          i + 1 > Array.length cousins_dates
+          || j + 1 > Array.length cousins_dates.(i)
+        then (-1, -1)
+        else cousins_dates.(i).(j)
+      in
       VVstring (string_of_int min)
   | [ "cous_paths_max_date"; l1; l2 ] ->
-      let max_a_l =
-        match get_env "max_anc_level" env with
-        | Vint i -> i
-        | _ -> max_anc_level_default
-      in
-      let max_d_l =
-        match get_env "max_desc_level" env with
-        | Vint i -> i
-        | _ -> max_desc_level_default
-      in
       let _cousins_cnt, cousins_dates =
         match (!Cousins.cousins_t, !Cousins.cousins_dates_t) with
         | Some t, Some d_t -> (t, d_t)
-        | _, _ -> Cousins.init_cousins_cnt base max_a_l max_d_l p
+        | _, _ -> Cousins.init_cousins_cnt conf base p
       in
       let i = try int_of_string l1 with Failure _ -> raise Not_found in
       let j = try int_of_string l2 with Failure _ -> raise Not_found in
-      let _, max = cousins_dates.(i).(j) in
+      let _, max =
+        if
+          i + 1 > Array.length cousins_dates
+          || j + 1 > Array.length cousins_dates.(i)
+        then (-1, -1)
+        else cousins_dates.(i).(j)
+      in
```

Comment:
I would prefer to have both `cous_paths_min_date` and `cous_paths_max_date` to be in Cousins module.
This way `cousins_t` and `cousins_dates_t` do not have to be exposed in Cousins interface.
And the rest of the code will not have to work on cousins_dates array and have to know how cousins and cousins dates computations works.

Reply:
Done

Reply:
thanks!